src/main/java/seedu/address/model/person/staff/ShiftRoster.java

```diff
@@ -0,0 +1,101 @@
+package seedu.address.model.person.staff;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Represents a collection of shifts for a single staff in the restaurant book.
+ *
+ * Guarantees: immutable;
```

Comment:
Given that you can add and remove shifts from the list, is it really immutable?

Reply:
The list of shifts in the ShiftRoster class is made unmodifiable in the constructor.
The add and delete methods make a copy of the list of shifts, adds/removes a shift and returns a new ShiftRoster object with the updated list of shifts.
In this sense the class is unmodifiable.